packages/rich-text/src/test/apply-format.js

```diff
@@ -49,4 +51,24 @@ describe( 'applyFormat', () => {
 		expect( result ).not.toBe( record );
 		expect( getSparseArrayLength( result.formats ) ).toBe( 4 );
 	} );
+
+	it( 'should apply format on existing format if selection is collapsed', () => {
+		const record = {
+			formats: [ , , , , [ a ], [ a ], [ a ], , , , , , , ],
+			text: 'one two three',
+			start: 4,
+			end: 4,
+		};
+		const expected = {
+			formats: [ , , , , [ a2 ], [ a2 ], [ a2 ], , , , , , , ],
+			text: 'one two three',
+			start: 4,
+			end: 4,
+		};
+		const result = applyFormat( deepFreeze( record ), a2 );
+
+		expect( result ).toEqual( expected );
+		expect( result ).not.toBe( record );
+		expect( getSparseArrayLength( result.formats ) ).toBe( 3 );
+	} );
```

Comment:
What happens if `applyFormat` is called with no existing format and a collapsed selection? Should we have a unit test for this case?

Reply:
Then nothing happens, but sure we can have a test for it.

Reply:
I will do it now and merge PR afterwards 👍

Reply:
I hope this works for you @noisysocks: https://github.com/WordPress/gutenberg/pull/10372/commits/9221a33e3a60535221e6d165d96b1385331411a2